pkg/containerfs/archiver.go

```diff
@@ -148,8 +148,6 @@ func (archiver *Archiver) CopyFileWithTar(src, dst string) (err error) {
 			hdr.Name = dstDriver.Base(dst)
 			if dstDriver.OS() == "windows" {
 				hdr.Mode = int64(chmodTarEntry(os.FileMode(hdr.Mode)))
-			} else {
-				hdr.Mode = int64(os.FileMode(hdr.Mode))
```

Comment:
What does it do?

Reply:
I am going to provide some tests. During my tests this was not working correctly. Actually this looks like no-op or something that will broke your permissions.

There is hdr.FileInfo().Mode() for accessing whole permissions and i don't understand the purpose of this statement.

Reply:
```.go
package main

import (
        "archive/tar"
        "fmt"
        "os"
)

func main() {
        srcSt, _ := os.Stat("1.bin")
        fmt.Printf("srcSt: %v\n", srcSt.Mode())
        hdr, _  := tar.FileInfoHeader(srcSt, "")

        fmt.Printf("hdr.Fileinfo().Mode(): %v\n", hdr.FileInfo().Mode())
}
```
```
ls -lah 1.bin 
-rwSrwSr-- 1 navern navern 0 ноя  9 20:01 1.bin

navern@errnil:~/tmp/bla/tmp$ ./1 
srcSt: ugrw-rw-r--
hdr.Fileinfo().Mode(): ugrw-rw-r-- 
```

You get correct mode after tar.FileInfoHeader, i dont understand why you need this line:
```
hdr.Mode = int64(os.FileMode(hdr.Mode))
```

Reply:
I've returned this line because it doesn't impact this particular change and i don't have full understanding of what's going on here)